core/resharding.md

```diff
@@ -0,0 +1,12 @@
+## Resharding
+
+In current design, the epoch is defined to be fixed length, the epoch length is a constant parameter BlocksPerEpoch. In future, it will be dynamically adjustable according to security parameter. During the epoch transition, suppose there are N shards, we sort the shards according to the size of active nodes (that had staking for next epoch). The first N/2 larger shards will be called active committees, and the last N/2 smaller shards will be called inactive committees. Don't be confused by
+the name, they are all normal shards with same functional.
+
+All the information about sharding will be stored in BeaconChain. A sharding state is defined as a map which maps each NodeID to the ShardID the node belongs to. Every node will have a unique NodeID and be mapped to one ShardID. At the beginning of a new epoch, the BeaconChain leader will propose a new block containing the new sharding state, the new sharding state is uniquely determined by the randomness generated by verifiable random function (VRF). During the consensus process, all the validators will perform the same calculation and verify the proposed sharding state is valid. After consensus reach, each node will write the new sharding state into the block. This block is called epoch block. In current code, it's the first block of each epoch in BeaconChain.
```

Comment:
generated by distributed randomness protocol (not just VRF)

Reply:
After consensus (is reached)

Reply:
thanks! corrected